bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -1411,4 +1418,50 @@ private void updateThingFromLegacy() {
             }
         }
     }
+
+    /**
+     * Update the thing's semantic equipment tag. The main determinant for the equipment type is the supported channels.
+     * For lights use the product archetype to split between light bulbs and strip lights. Rooms and Zones are
+     * considered to be (groups of) light bulbs.
+     */
+    private void updateEquipmentTag() {
+        if (!disposing) {
+            int sensorCount = 0;
+            SemanticTag equipmentTag = Equipment.LIGHTBULB; // default;
+
+            if (Set.of(ResourceType.ROOM, ResourceType.ZONE).contains(thisResource.getType())) {
+                equipmentTag = Equipment.ZONE;
+            }
+            if (thisResource.getProductData() instanceof ProductData productData
+                    && STRIPLIGHT_ARCHETYPES.contains(productData.getProductArchetype())) {
+                equipmentTag = Equipment.LIGHT_STRIP;
+            }
+            if (thing.getChannel(CHANNEL_2_BUTTON_LAST_EVENT) != null) {
```

Comment:
I'm wondering if there is a more direct way to obtain the `ResourceType` for the Thing? From a quick look I guess `thisResource` is only guaranteed to be either `DEVICE`, `ROOM` or `ZONE`? If we could ensure that we have the specialized resource type, we could avoid the dependency/indirection of first having configured the channels. Which I guess happens after receiving a resource for which the type is checked? I don't remember how it all works, so just had a brief look.

Reply:
The fundamental resource types are Device, Room, Zone. The exact kind of Device depends on which services the thing supports, and the list of channels is reduced from all channels down to the subset of channels that corespond to a respective service. I suppose we could duplicate the code that does the channel reduction also to do the equipment tag setting. However rather than duplicating code, I think it is better to leverage that code to therefore also determine the tag.

Reply:
I wouldn't propose to duplicate logic, but rather extract it for reusing it. If you don't think that's feasible, we can leave it as is.

Reply:
I don't have the code in front of me, but as I recall the updateTags code is called directly after the updateChannels. Of course we could include the code of the former method in the latter method, but that slightly goes against the code to reduce method sizes by extracting sub functions into sub methods..

Reply:
See the code at line 1074 
```
https://github.com/openhab/openhab-addons/blob/cd71ab5f8ec6337109311cca6782b7b52b13b50c/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java#L1062
```

Reply:
> a more direct way to obtain the ResourceType for the Thing? 
> if we could avoid the dependency/indirection of first having configured the channels

As discussed above, the answer is no. The function of a Device Thing depends on the services that it supports. The initialization code `updateDependencies()` method calls `updateLookups()`, `updateServiceContributors()`, `updateChannelList()`, `updateChannelItemLinksFromLegacy()`, and finally `updateEquipmentTag()` in direct sequence. There is no more direct manner than that.

So I am closing this issue as resolved.